compiler/optimizer/ValuePropagationCommon.cpp

```diff
@@ -922,6 +922,47 @@ bool OMR::ValuePropagation::transformUnsafeCopyMemoryCall(TR::Node *arrayCopyNod
    return false;
    }
 
+static void setNodeOnList(TR::Node *currNode, TR_BitVector *nodeList)
+   {
+   if (nodeList->isSet(currNode->getGlobalIndex()))
+      return;
+
+   nodeList->set(currNode->getGlobalIndex());
+
+   for (int32_t i = 0; i < currNode->getNumChildren(); i++)
+      {
+      setNodeOnList(currNode->getChild(i), nodeList);
+      }
+   }
+
+static void findAndReplaceCommonedNodeWithLoadFromTemp(TR::Node *currNode, TR::Node *matchingNode, TR::SymbolReference *symRef, TR_BitVector *nodeSkipList, bool trace, TR::Compilation *comp)
```

Comment:
May I ask you to add some Doxygen comments or at least some description of what it's doing? I know there's a bit of a description that appears where it's called, but I think it would help to have it described here.